Source/WebCore/dom/MessagePort.cpp

```diff
@@ -350,14 +350,17 @@ bool MessagePort::virtualHasPendingActivity() const
     if (!context || m_isDetached)
         return false;
 
-    // If this object has been idle since the remote port declared itself elgibile for GC, we can GC.
-    if (!m_hasHadLocalActivitySinceLastCheck && m_isRemoteEligibleForGC)
-        return false;
-
     // If this MessagePort has no message event handler then the existence of remote activity cannot keep it alive.
     if (!m_hasMessageEventListener)
         return false;
 
+    if (m_entangled)
```

Comment:
I think it would be slightly clearer to check isEntangled() here and drop the check for m_isDetached above.

Reply:
But it wouldn't be equivalent / right as far as I can tell.

The current code:
- Return false if detached.
- Return true if entangled.
- Then returns true/false based on other flags

With your proposal:
- Return true if entangled and not detached
- Then returns true/false based on other flags

Because we would no longer early false when m_detached, it would be possible (at least in theory) for this function to return true while m_detached=true, based on the other flags. I believe this would be wrong.